kobuki_safety_controller/launch/safety_controller-composed-launch.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+# All rights reserved.
+#
+# Software License Agreement (BSD License 2.0)
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+# * Redistributions of source code must retain the above copyright
+#   notice, this list of conditions and the following disclaimer.
+# * Redistributions in binary form must reproduce the above
+#   copyright notice, this list of conditions and the following
+#   disclaimer in the documentation and/or other materials provided
+#   with the distribution.
+# * Neither the name of {copyright_holder} nor the names of its
+#   contributors may be used to endorse or promote products derived
+#   from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+"""Launch the safety controller node composed with default configuration."""
+
+import os
+
+import ament_index_python.packages
+
+from launch import LaunchDescription
+from launch_ros.actions import ComposableNodeContainer
+from launch_ros.descriptions import ComposableNode
+
+import yaml
+
+
+def generate_launch_description():
+    share_dir = ament_index_python.packages.get_package_share_directory('kobuki_safety_controller')
+    params_file = os.path.join(share_dir, 'config', 'safety_controller_params.yaml')
+    with open(params_file, 'r') as f:
+        params = yaml.safe_load(f)['kobuki_safety_controller_node']['ros__parameters']
+    container = ComposableNodeContainer(
+            node_name='safety_controller_container',
+            node_namespace='',
+            package='rclcpp_components',
+            node_executable='component_container',
+            composable_node_descriptions=[
+                ComposableNode(
+                    package='kobuki_safety_controller',
+                    node_plugin='kobuki_safety_controller::SafetyController',
+                    node_name='safety_controller_node',
+                    parameters=[params]),
+            ],
+            output='both',
```

Comment:
Curious, what does 'both' mean here?

Reply:
Yeah, this is one of the bits about launch I don't like.  `both` means to both log to the screen and to log to a logfile (by default to a directory in `~/.ros/log`).